scripts/internal/migrate_billing_cycle.go

```diff
@@ -0,0 +1,221 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/flexprice/flexprice/internal/cache"
+	"github.com/flexprice/flexprice/internal/config"
+	domainSub "github.com/flexprice/flexprice/internal/domain/subscription"
+	"github.com/flexprice/flexprice/internal/logger"
+	"github.com/flexprice/flexprice/internal/postgres"
+	"github.com/flexprice/flexprice/internal/repository/ent"
+	"github.com/flexprice/flexprice/internal/sentry"
+	"github.com/flexprice/flexprice/internal/types"
+)
+
+// MigrateBillingCycleParams holds parameters for billing cycle migration
+type MigrateBillingCycleParams struct {
+	TenantID      string
+	EnvironmentID string
+	DryRun        bool
+}
+
+// MigrateBillingCycle migrates subscriptions from anniversary to calendar billing cycle
+func MigrateBillingCycle() error {
```

Comment:
Add a huge comment critical warkin - does not handle proration for subscriptions when changing the dates